File: src/main/java/com/sport/service/web/models/training_program/CreateTrainingProgramRequest.java
package com.sport.service.web.models.training_program;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class CreateTrainingProgramRequest {
    @NotEmpty(message = "Coaches' ids must be pointed")
    private List<Long> coachesId;
    @NotBlank(message = "Title must be pointed")
    private String title;
    private BigDecimal price;
    private String description;
    @NotEmpty(message = "Sport types must be pointed")
    private List<String> sportTypes;
    private Boolean showInWeb;
    private Boolean haveVideoMaterials;
    private String contactToBuy;
}

File: src/main/kotlin/com/sport/service/web/models/training_program/CreateTrainingProgramRequest.kt
package com.sport.service.web.models.training_program

import jakarta.validation.constraints.NotBlank
import java.math.BigDecimal

data class CreateTrainingProgramRequest(
    @field:NotBlank(message = "Coach's/es' name/s must be pointed")
    val coachesId: List<Long>,
    @field:NotBlank(message = "Title must be pointed")
    val title: String,
    val price: BigDecimal,
    val description: String,
    val sportTypes: List<String>,
    val showInWeb: Boolean
)